python_modules/dagster/dagster/_daemon/sensor.py

```diff
@@ -638,8 +639,12 @@ def _evaluate_sensor(
                     asset_selection=stale_assets, stale_assets_only=False
                 )
 
-        target_data: ExternalTargetData = check.not_none(
-            external_sensor.get_target_data(run_request.job_name)
+        target_data: Optional[ExternalTargetData] = external_sensor.get_target_data(
+            run_request.job_name
+        ) or ExternalTargetData(
```

Comment:
nits:
prefer explicit if / else-ing w/ comments here to make it clear what the assumptions we are operating against are

might be clearer to just move away from `ExternalTargetData` and move this handling in to `PipelineSelector` construction